components/AllProducts/pagination.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from "react";
+import styles from "./allProduct.module.css";
+
+// props에 대한 타입 정의
+interface PaginationProps {
+  currentPage: number; // 현재 페이지 번호
+  totalItems: number; // 전체 아이템 개수
+  itemsPerPage: number; // 페이지당 아이템 개수
+  onPageChange: (page: number) => void; // 페이지 변경 시 호출되는 함수
+}
+
+const pageRange = 5; // 한 번에 보여줄 페이지 버튼 개수
```

Comment:
상수는 대문자로 적어주시는 걸 추천드려요!
```suggestion
const PAGE_RANGE = 5; // 한 번에 보여줄 페이지 버튼 개수
```